tests/test_stdin_timeout.py

```diff
@@ -0,0 +1,31 @@
+from wandb.sdk.lib.stdin_timeout import stdin_timeout
+from wandb.errors import InputTimeoutError
+from wandb.util import prompt_choices
+import io
+import pytest
+
+
+@pytest.fixture
+def stdin_timeout(monkeypatch):
+    monkeypatch.setattr("sys.stdin", io.StringIO("1"))
+    choice = stdin_timeout("", None, "")
+    assert choice == "1"
+
+    choices = ["1", "2"]
+    choice = prompt_choices(choices)
+    assert choice == "1"
+
+
+"""
+@pytest.fixture
+def input_timeout_no_choice(pytestconfig):
+    capmanager = pytestconfig.pluginmanager.getplugin('capturemanager')
+    capmanager.suspend_global_capture(_in=True)
+
+    timeout_log = "input timeout!"
+    try:
+        stdin_timeout("waiting for input", 1, timeout_log)
+    except InputTimeoutError as e:
+        assert e.message == timeout_log
+    capmanager.resume_global_capture()
+"""
```

Comment:
What's up with this test?

Reply:
This is a hacky way of enabling stdin in pytest. It's not recommended to use because these low level pytest APIs can change with updates.

The bigger issue here is that the tests in this file are not being run by in tox. That's why codecov is failing. running `ox -e py37 -- tests/test_stdin_timeout.py` says that no tests were run and codecov didn't pickup any data. Do I need to put this file somewhere in the configs for tests to pick this up?
![Screenshot from 2021-08-14 12-40-11](https://user-images.githubusercontent.com/15766192/129438350-e60e39b0-900f-44ec-9e6f-8d0ca7d80170.png)

Reply:
pytest determines which functions are tests by looking for method names prepended with "test_".  Just change your cases to be named like all of the other test cases.

Reply:
@vanpelt got it thanks. Updated the test to only check timeout enabled cases. Windows test passes and gets covered in codecov. But linux/mac tests fail with a permission errors. Again, this only happens with pytest. login timeout works fine in manual tests. 
![Screenshot (78)](https://user-images.githubusercontent.com/15766192/129619100-f5f3fa44-c471-4a18-81ce-1484c3d64d9c.png)

Reply:
I imagine this is not related to pytest, instead it's related to linux or docker default permissions.  I would do some googling to see if you can find any other examples of PermissionError when registering Epoll selectors.

Reply:
After some investigation, I think it's related to pytest not allowing some os operations. suspending cap manager enables stdin but registering a selector might require some other workaround. 
I've tested the same test with pytest as the root user. It gets the same `operation not allowed` error. I've asked this on pytest forum but I'm not sure how swift their response will be.
Another way is to test this part of code independent of pytest, like a python script. Is it possible under current settings?

Reply:
@vanpelt ^^

Reply:
You should be able to use the new yea [functional testing framework ](https://github.com/wandb/client/tree/master/functional_tests) to test this.  I'm not super familiar with it, but it should be pretty straight forward after looking at the examples.  It may be tricky to convince yea that you have a tty.  The other option is to just launch a subprocess that runs code from within `/fixtures` and makes assertions based on the stdout but you won't get credit for the coverage without using the functional_tests